crates/rattler_networking/src/authentication_storage/storage.rs

```diff
@@ -115,10 +115,29 @@ impl AuthenticationStorage {
             let credentials = match credentials {
                 Ok(None) => {
                     // Check for credentials under e.g. `*.prefix.dev`
-                    let mut parts = host.rsplitn(2, '.').collect::<Vec<&str>>();
-                    parts.reverse();
-                    let wildcard_host = format!("*.{}", parts.join("."));
-                    self.get(&wildcard_host)
+                    let domain = match url.domain() {
```

Comment:
Perfect usecase for let Some(domain) = ... else { return ... } syntax!

(Sorry im on a phone)

Reply:
I wanted the early return instead of nesting everything and i think the match syntax is cleaner for this

Reply:
I think you misunderstand, I mean this syntax: https://doc.rust-lang.org/beta/rust-by-example/flow_control/let_else.html. 

Its relatively new syntax similar to a match statement but easier to read in cases like this. It also allows an early return.

Reply:
Wow, that's some nifty syntax, thanks!